### ISSUE TITLE
💅 False negative in useImportType when combining default and named imports

### ISSUE THREAD

**Issue by remcohaszing** [biomejs/biome#10493]
### Environment information

<details>

```bash
CLI:
  Version:                      2.3.11
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.15.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false
  HTML full support enabled:    unset

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  true
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noBannedTypes
    complexity/noCommaOperator
    complexity/noEmptyTypeParameters
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noForEach
    complexity/noImportantStyles
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringRaw
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/useArrowFunction
    complexity/useDateNow
    complexity/useFlatMap
    complexity/useIndexOf
    complexity/useLiteralKeys
    complexity/useNumericLiterals
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noMissingVarFunction
    correctness/noNestedComponentDefinitions
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useGraphqlNamedOperations
    correctness/useIsNan
    correctness/useJsxKeyInIterable
    correctness/useParseIntRadix
    correctness/useUniqueElementIds
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield
    performance/noAccumulatingSpread
    performance/noDynamicNamespaceImportAccess
    security/noBlankTarget
    security/noDangerouslySetInnerHtml
    security/noGlobalEval
    style/noDescendingSpecificity
    style/noInferrableTypes
    style/noNonNullAssertion
    style/noParameterAssign
    style/noUnusedTemplateLiteral
    style/noUselessElse
    style/useArrayLiterals
    style/useAsConstAssertion
    style/useConst
    style/useDefaultParameterLast
    style/useDeprecatedReason
    style/useEnumInitializers
    style/useExponentiationOperator
    style/useExportType
    style/useImportType
    style/useLiteralEnumMembers
    style/useNodejsImportProtocol
    style/useNumberNamespace
    style/useSelfClosingElements
    style/useShorthandFunctionType
    style/useSingleVarDeclarator
    style/useTemplate
    suspicious/noApproximativeNumericConstant
    suspicious/noArrayIndexKey
    suspicious/noAssignInExpressions
    suspicious/noAsyncPromiseExecutor
    suspicious/noBiomeFirstException
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConstEnum
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDocumentCookie
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateProperties
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noDuplicateTestHooks
    suspicious/noEmptyBlock
    suspicious/noEmptyInterface
    suspicious/noExplicitAny
    suspicious/noExportsInTest
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noGlobalIsNan
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noNonNullAssertedOptionalChain
    suspicious/noOctalEscape
    suspicious/noPrototypeBuiltins
    suspicious/noQuickfixBiome
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noThenProperty
    suspicious/noTsIgnore
    suspicious/noUnknownAtRules
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noUselessEscapeInString
    suspicious/noUselessRegexBackrefs
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useAwait
    suspicious/useBiomeIgnoreFolder
    suspicious/useDefaultSwitchClauseLast
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useIsArray
    suspicious/useIterableCallbackReturn
    suspicious/useNamespaceKeyword

Workspace:
  Open Documents:               0
```

</details>

### Rule name

useImportType

### Playground link

https://biomejs.dev/playground/?lintRules=useImportType&code=aQBtAHAAbwByAHQAIABhAGEALAAgAHsAIAB0AHkAcABlACAAYQBiACwAIAB0AHkAcABlACAAYQBjACAAfQAgAGYAcgBvAG0AIAAnAGEAJwAKAAoAaQBtAHAAbwByAHQAIABiAGEALAAgAHsAIABiAGIALAAgAHQAeQBwAGUAIABiAGMAIAB9ACAAZgByAG8AbQAgACcAYgAnAAoA

### Expected result

When `useImportyType` is configured with:

```json
{
  "linter": {
    "rules": {
      "style": {
        "useImportimportType": {
          "level": "error",
          "options": {
            "style": "separatedType"
          }
        }
      }
    }
  }
}
```

If an import statement has a default import and named type imports, and it does not have regular named imports, then the type imports are not reported.

So the rule behaves as expected for these:

```js
import a, { b, type c } from 'a'
import { type b, type c } from 'a'
```

But it doesn’t report:

```js
import a, { type b, type c } from 'a'
```

The playground link doesn’t show this, because I didn’t see a way to configure rule options.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Here's a better playground that contains the correct configuration

https://biomejs.dev/playground/?lintRules=useImportType&analyzerFixMode=safeAndUnsafeFixes&tab=formatter&pane=Console&files.main.tsx=aQBtAHAAbwByAHQAIABhAGEALAAgAHsAIAB0AHkAcABlACAAYQBiACwAIAB0AHkAcABlACAAYQBjACAAfQAgAGYAcgBvAG0AIAAnAGEAJwAKAGkAbQBwAG8AcgB0ACAAYgBhACwAIAB7ACAAYgBiACwAIAB0AHkAcABlACAAYgBjACAAfQAgAGYAcgBvAG0AIAAnAGIAJwAKAAoAaQBtAHAAbwByAHQAIABhACwAIAB7ACAAYgAsACAAdAB5AHAAZQAgAGMAIAB9ACAAZgByAG8AbQAgACcAYQAnAAoAaQBtAHAAbwByAHQAIAB7ACAAdAB5AHAAZQAgAGIALAAgAHQAeQBwAGUAIABjACAAfQAgAGYAcgBvAG0AIAAnAGEAJwAKAA%3D%3D&files.biome.json=ewAKACAAIAAiAGwAaQBuAHQAZQByACIAOgAgAHsACgAgACAAIAAgACIAcgB1AGwAZQBzACIAOgAgAHsACgAgACAAIAAgACAAIAAiAHMAdAB5AGwAZQAiADoAIAB7AAoAIAAgACAAIAAgACAAIAAgACIAdQBzAGUASQBtAHAAbwByAHQAVAB5AHAAZQAiADoAIAB7AAoAIAAgACAAIAAgACAAIAAgACAAIAAiAGwAZQB2AGUAbAAiADoAIAAiAGUAcgByAG8AcgAiACwACgAgACAAIAAgACAAIAAgACAAIAAgACIAbwBwAHQAaQBvAG4AcwAiADoAIAB7AAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIgBzAHQAeQBsAGUAIgA6ACAAIgBzAGUAcABhAHIAYQB0AGUAZABUAHkAcABlACIACgAgACAAIAAgACAAIAAgACAAIAAgAH0ACgAgACAAIAAgACAAIAAgACAAfQAKACAAIAAgACAAIAAgAH0ACgAgACAAIAAgAH0ACgAgACAAfQAKAH0A
